docs/developer-docs/latest/development/typescript.md

```diff
@@ -4,110 +4,13 @@ description: Learn how you can use Typescript for your Strapi application.
-| `./dist` directory                        | Adds the location for compiling the project JavaScript source code.                                                                               | application root |
-| `build` directory                         | Contains the compiled administration panel JavaScript source code.  The directory is created on the first `yarn build` or `npm run build` command | `./dist/`         |
-| `tsconfig.json` file                      | Manages TypeScript compilation for the server.                                                                                                    | application root |
-| `tsconfig.json` file                      | Manages TypeScript compilation for the admin panel.                                                                                               | `./src/admin/`   |
-
-Starting the development environment for a TypeScript-enabled project requires building the admin panel prior to starting the server. In development mode, the application source code is compiled to the `./dist/build` directory and recompiled with each change in the Content-type Builder. To start the application, run the following commands in the root directory:
-
-<code-group>
-
-<code-block title="NPM">
-
-```sh
-npm run build
-npm run develop
-```
-
-</code-block>
-
- <code-block title="YARN">
-
-```sh
-yarn build
-yarn develop
-```
-
-</code-block>
-
-</code-group>
-
-## Use TypeScript typings
-
-Strapi provides typings on the Strapi class to improve the TypeScript experience. In the following example a `register` method is added to `./src/index.ts`. For example, when the `strapi.runLifecyclesFunctions` method is added, a list of available lifecycle types (`register`, `bootstrap` and `destroy`) are returned by the code editor. The following example is illustrated statically and the second tab shows an animated example.
-
-:::: tabs card
-
-::: tab STATIC EXAMPLE
-
-```ts
-// path: ./src/index.ts
-
-import '@strapi/strapi';
-
-export default {
-
-  register( {strapi }: { strapi: Strapi }) {
-    strapi.runLifecyclesFunctions('register') .then(()  => console.log('hello world')
-    )
-    
-  },
-};
-```
-
-:::
-
-::::
-
+TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.2.0 and above. The [CLI documentation](/developer-docs/latest/setup-deployment-guides/installation/cli.md) describes how to develop a new TypeScript project. Additionally, the [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md) and [optional configurations](/developer-docs/latest/setup-deployment-guides/configurations/optional/typescript.md) sections have TypeScript-specific resources. The core Developer Documentation contains code examples in both JavaScript and TypeScript.
```

Comment:
```suggestion
TypeScript adds an additional type system layer above JavaScript, which means that existing JavaScript code is also TypeScript code. Strapi supports TypeScript in new projects on v4.2.0 and above. The [CLI documentation](/developer-docs/latest/setup-deployment-guides/installation/cli.md) describes how to create a new TypeScript project. Additionally, the [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md) and [optional configurations](/developer-docs/latest/setup-deployment-guides/configurations/optional/typescript.md) sections have TypeScript-specific resources. The core Developer Documentation contains code examples in both JavaScript and TypeScript.
```

To me, the CLI section only explains how to create a TS project. To _develop_ one, people will need to follow the instructions we'll give in the docs-next content.
Also, I'd be a bit more specific in the other sentences. For instance, the TS configuration (config > optional > typescript page) is "optional" from the global point of view of a Strapi app (if you don't use TS, you don't need this config.), but is an absolute requirement to start developing Strapi with TS, or the code won't compile.

I think I would even use a prerequisites callout mentioning that before starting developing with TS you either need to create a new project with the appropriate CLI or at least add the required config files. (to be checked with devs).

Reply:
- I updated the sentences. From the existing callout and the written text, we are not suggesting or providing any methods for converting existing projects to TypeScript projects. The only provided method is to use the CLI to make a new project with the appropriate flag. Eventually, there will be instructions for incorporating TypeScript into an existing project but that is not ready. 
- I used "core" because I don't think we checked or altered lots of the vestigial documentation for TypeScript (I am thinking of the old guides) but I am fine with the word removed as well. 
- I added the prerequisites callout, but now the callout and the introduction are duplicating content so I am not sure what you intended for the final result?